exonum-java-binding/core/src/main/java/com/exonum/binding/core/transaction/Preconditions.java

```diff
@@ -0,0 +1,265 @@
+/*
+ * Copyright 2020 The Exonum Team
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.exonum.binding.core.transaction;
+
+import static com.google.common.base.Strings.lenientFormat;
+
+import org.checkerframework.checker.nullness.qual.Nullable;
+
+/**
+ * Utility methods that helps verifying conditions conducted in expression
+ * while transaction execution.
+ * If the condition is not met, the {@code Preconditions} method throws {@link ExecutionException}.
+ * <p>Consider the following example:
+ * <pre>{@code
+ *   void checkEnoughMoney(long balance, long amount) {
+ *     if(balance < amount) {
+ *       throw new ExecutionException((byte)3, "Not enough money. Operation amount is " + amount
+ *       + ", but actual balance was " + balance);
+ *     }
+ *   }
+ * }</pre>
+ * <p>which can be replaced using Preconditions:
+ * <pre>{@code
+ *   check(amount <= balance, (byte)3,
+ *       "Not enough money. Operation amount is %s, but actual balance was %s",
+ *       amount, balance);
+ * }</pre>
+ */
+public final class Preconditions {
+
+  /**
+   * Verifies the truth of the given expression.
+   *
+   * @param expression a boolean expression
+   * @param errorCode execution error code
+   * @throws ExecutionException if {@code expression} is false
+   */
+  public static void check(boolean expression, byte errorCode) {
```

Comment:
there are 2 questions: 
- do we need to add `int errorCode` overload?
- do we need to validate errorCode values?